sdk/storage/Azure.Storage.Blobs/src/Models/BlobUploadDirectoryOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// Blob Upload Directory Options
+    /// <see cref="BlobUploadOptions"/> for more options.
+    /// </summary>
+    public class BlobUploadDirectoryOptions : BlobUploadOptions
```

Comment:
I'd rather not inherit from BlobUploadOptions. Let's not couple these types together in case we want to evolve them separately.

If there's commonality I'd rather use internal interfaces to keep common part in consistent and reduce duplication in internal components consuming this.